File: common/src/test/java/com/hartwig/actin/molecular/sort/driver/DriverLikelihoodComparatorTest.java
package com.hartwig.actin.molecular.sort.driver;

import static org.junit.Assert.assertEquals;

import java.util.List;

import com.google.common.collect.Lists;
import com.hartwig.actin.molecular.datamodel.driver.DriverLikelihood;

import org.junit.Test;

public class DriverLikelihoodComparatorTest {

    @Test
    public void canSortDriverLikelihoods() {
        DriverLikelihood high = DriverLikelihood.HIGH;
        DriverLikelihood medium = DriverLikelihood.MEDIUM;
        DriverLikelihood low = DriverLikelihood.LOW;
        DriverLikelihood nothing = null;

        List<DriverLikelihood> driverLikelihoods = Lists.newArrayList(medium, low, high, nothing);
        driverLikelihoods.sort(new DriverLikelihoodComparator());

        assertEquals(high, driverLikelihoods.get(0));
        assertEquals(medium, driverLikelihoods.get(1));
        assertEquals(low, driverLikelihoods.get(2));
        assertEquals(nothing, driverLikelihoods.get(3));
    }
}

File: common/src/test/kotlin/com/hartwig/actin/molecular/sort/driver/DriverLikelihoodComparatorTest.kt
package com.hartwig.actin.molecular.sort.driver

import com.hartwig.actin.molecular.datamodel.DriverLikelihood
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class DriverLikelihoodComparatorTest {

    @Test
    fun `Should sort driver likelihoods`() {
        val high = DriverLikelihood.HIGH
        val medium = DriverLikelihood.MEDIUM
        val low = DriverLikelihood.LOW
        val nothing: DriverLikelihood? = null
        val driverLikelihoods = listOf(medium, low, high, nothing).sortedWith(DriverLikelihoodComparator())

        assertThat(driverLikelihoods[0]).isEqualTo(high)
        assertThat(driverLikelihoods[1]).isEqualTo(medium)
        assertThat(driverLikelihoods[2]).isEqualTo(low)
        assertThat(driverLikelihoods[3]).isEqualTo(nothing)
    }
}